driver-async/src/main/com/mongodb/async/client/MongoClients.java

```diff
@@ -213,9 +215,10 @@ public static CodecRegistry getDefaultCodecRegistry() {
 
     private static final CodecRegistry DEFAULT_CODEC_REGISTRY =
             fromProviders(asList(new ValueCodecProvider(),
-                    new DocumentCodecProvider(),
                     new BsonValueCodecProvider(),
-                    new IterableCodecProvider(),
+                    new DBRefCodecProvider(),
+                    new DocumentCodecProvider(new DocumentToDBRefTransformer()),
```

Comment:
This is the crux of the backward compatibility break.  An application that are currently decoding dbrefs into documents are getting those dbrefs as normal sub-documents (as with BsonDocument).  Now it will get them as DBRef instances instead.  An alternative is to require the application to opt in to this behavior by creating their own registry, which it could do now without taking a dependency on driver (based on moving everything to core).  This is similar to what we require for users to opt in to any other decoding behavior, e.g. changing from Date to Instant when decoding dates.

What do you think?